### PR TITLE
fix: update admin gui path

### PR DESCRIPTION
**Pull request by esoubiran-aneo**
This PR update the admin gui path to have the first (0.8) and the second gui (0.9) in the same deployment.

I also add deprecated message since GUI 0.8 and 0.9 will have been deprecated in our repo.

_Maybe, there is some image tag to change but I don't find where. But I think it's only in the ArmoniK repo that we have to do these change._

### CODE REVIEW

File: armonik/external-data.tf [hunk 1]
```diff
@@ -38,10 +45,17 @@ locals {
...
+  admin_0_8_gui_endpoints = {
+    ip       = module.admin_0_8_gui_endpoint.host
+    api_port = try(module.admin_0_8_gui_endpoint.ports[0], null)
+    app_port = try(module.admin_0_8_gui_endpoint.ports[1], null)
+  }
+  # Deprecated, must be removed in a future version
+  admin_0_9_gui_endpoints = {
+    ip       = module.admin_0_9_gui_endpoint.host
+    api_port = try(module.admin_0_9_gui_endpoint.ports[0], null)
+    app_port = try(module.admin_0_9_gui_endpoint.ports[1], null)
```
Comment (lemaitre-aneo) [inline armonik/external-data.tf:58]: I think 0.9 has only the app port:
```suggestion
  admin_0_9_gui_endpoints = {
    ip       = module.admin_0_9_gui_endpoint.host
    port = try(module.admin_0_9_gui_endpoint.ports[0], null)
```

File: armonik/external-data.tf [hunk 2]
```diff
@@ -51,8 +65,12 @@ locals {
...
   admin_app_url     = length(kubernetes_service.admin_gui) > 0 ? "http://${local.admin_gui_endpoints.ip}:${local.admin_gui_endpoints.app_port}" : null
-  admin_api_url     = length(kubernetes_service.admin_old_gui) > 0 ? "http://${local.admin_old_gui_endpoints.ip}:${local.admin_old_gui_endpoints.api_port}/api" : null
-  admin_old_url     = length(kubernetes_service.admin_old_gui) > 0 ? "http://${local.admin_old_gui_endpoints.ip}:${local.admin_old_gui_endpoints.app_port}/" : null
+  # Deprecated, must be removed in a future version
+  admin_api_url     = length(kubernetes_service.admin_0_8_gui) > 0 ? "http://${local.admin_0_8_gui_endpoints.ip}:${local.admin_0_8_gui_endpoints.api_port}/api" : null
+  # Deprecated, must be removed in a future version
+  admin_0_8_url     = length(kubernetes_service.admin_0_8_gui) > 0 ? "http://${local.admin_0_8_gui_endpoints.ip}:${local.admin_0_8_gui_endpoints.app_port}/" : null
+  # Deprecated, must be removed in a future version
+  admin_0_9_url     = length(kubernetes_service.admin_0_9_gui) > 0 ? "http://${local.admin_0_9_gui_endpoints.ip}:${local.admin_0_9_gui_endpoints.app_port}/" : null
```
Comment (lemaitre-aneo) [inline armonik/external-data.tf:73]:
```suggestion
  admin_0_9_url     = length(kubernetes_service.admin_0_9_gui) > 0 ? "http://${local.admin_0_9_gui_endpoints.ip}:${local.admin_0_9_gui_endpoints.port}/" : null
```

File: armonik/ingress-configmap.tf [hunk 3]
```diff
@@ -65,19 +65,35 @@ server {
...
+        rewrite ^ $scheme://$http_host/admin-0.8/ permanent;
+    }
+    # Deprecated, must be removed in a new version
+    location = /admin-0.8 {
+        rewrite ^ $scheme://$http_host/admin-0.8/ permanent;
+    }
+    # Deprecated, must be removed in a new version
+    location = /admin-0.9 {
+        rewrite ^ $scheme://$http_host/admin-0.9/ permanent;
     }
```
Comment (lemaitre-aneo) [inline armonik/ingress-configmap.tf:79]: I think you would also need the language redirection for 0.9 as for `admin`.

File: armonik/variables.tf [hunk 4]
```diff
@@ -164,8 +164,50 @@ variable "admin_gui" {
...
+    replicas           = number
+    image_pull_policy  = string
+    image_pull_secrets = string
+    node_selector      = any
+  })
+  default = null
+}
+
+# Deprecated, must be removed in a future version
+variable "admin_0_9_gui" {
```
Comment (lemaitre-aneo) [inline armonik/variables.tf:210]: The parameters of 0.9 should be a copy of the ones from `admin_gui`, not `admin_old_gui`.

File: armonik/variables.tf [hunk 5]
```diff
@@ -164,8 +164,50 @@ variable "admin_gui" {
...
+      limits = object({
+        cpu    = string
+        memory = string
+      })
+      requests = object({
+        cpu    = string
+        memory = string
+      })
+    })
+    old = object({
```
Comment (lemaitre-aneo) [inline armonik/variables.tf:186]: I know that is was already there, but I think this parameter should be renamed to `app`. We are already in a `admin_0_8_gui` parameter.

```suggestion
    app = object({
```

File: armonik/admin-0-8-gui.tf [hunk 6]
```diff
@@ -1,29 +1,29 @@
 # Control plane deployment
-resource "kubernetes_deployment" "admin_old_gui" {
-  count = var.admin_old_gui != null ? 1 : 0
+resource "kubernetes_deployment" "admin_0_8_gui" {
+  count = var.admin_0_8_gui != null ? 1 : 0
   metadata {
-    name      = "admin-old-gui"
+    name      = "admin-0_8-gui"
```
Comment (lemaitre-aneo) [inline armonik/admin-0-8-gui.tf:5]: nitpick: I would prefer dash:
```suggestion
    name      = "admin-0-8-gui"
```

File: armonik/admin-0-8-gui.tf [hunk 7]
```diff
@@ -106,17 +106,17 @@ resource "kubernetes_deployment" "admin_old_gui" {
...
         }
-        # Old GUI container
+        # 0_8 GUI container
         container {
-          name              = var.admin_old_gui.old.name
-          image             = var.admin_old_gui.old.tag != "" ? "${var.admin_old_gui.old.image}:${var.admin_old_gui.old.tag}" : var.admin_old_gui.old.image
-          image_pull_policy = var.admin_old_gui.image_pull_policy
+          name              = var.admin_0_8_gui.0_8.name
+          image             = var.admin_0_8_gui.0_8.tag != "" ? "${var.admin_0_8_gui.0_8.image}:${var.admin_0_8_gui.0_8.tag}" : var.admin_0_8_gui.0_8.image
+          image_pull_policy = var.admin_0_8_gui.image_pull_policy
```
Comment (lemaitre-aneo) [inline armonik/admin-0-8-gui.tf:113]:
```suggestion
          name              = var.admin_0_8_gui.app.name
          image             = var.admin_0_8_gui.app.tag != "" ? "${var.admin_0_8_gui.app.image}:${var.admin_0_8_gui.app.tag}" : var.admin_0_8_gui.app.image
          image_pull_policy = var.admin_0_8_gui.image_pull_policy
```

File: armonik/admin-0-8-gui.tf [hunk 8]
```diff
@@ -106,17 +106,17 @@ resource "kubernetes_deployment" "admin_old_gui" {
...
           resources {
-            limits   = var.admin_old_gui.old.limits
-            requests = var.admin_old_gui.old.requests
+            limits   = var.admin_0_8_gui.0_8.limits
+            requests = var.admin_0_8_gui.0_8.requests
```
Comment (lemaitre-aneo) [inline armonik/admin-0-8-gui.tf:116]:
```suggestion
            limits   = var.admin_0_8_gui.app.limits
            requests = var.admin_0_8_gui.app.requests
```

File: armonik/admin-0-9-gui.tf
Comment (lemaitre-aneo) [inline armonik/admin-0-9-gui.tf:1]: This file should have been copied from admin-gui.tf, not admin-old-gui.tf.
Specifically: Gui 0.9 does not have an API container.

**Review by lemaitre-aneo — CHANGES_REQUESTED** [review]

File: armonik/external-data.tf (hunk 1 above)
Comment (esoubiran-aneo) [reply armonik/external-data.tf:58]: You're right!

File: armonik/external-data.tf (hunk 2 above)
Comment (esoubiran-aneo) [reply armonik/external-data.tf:72]: Do we need to also update line 46 to be consistent? (from app_port to port)

File: armonik/ingress-configmap.tf (hunk 3 above)
Comment (esoubiran-aneo) [reply armonik/ingress-configmap.tf:79]: Yes, I thinks it could be useful.

File: armonik/variables.tf (hunk 4 above)
Comment (esoubiran-aneo) [reply armonik/variables.tf:210]: Yes, I change it.

File: armonik/admin-0-9-gui.tf
Comment (esoubiran-aneo) [reply armonik/admin-0-9-gui.tf:1]: Yes, I do it wrong.

File: armonik/admin-0-8-gui.tf (hunk 6 above)
Comment (esoubiran-aneo) [reply armonik/admin-0-8-gui.tf]: Should I change it in every name where a dash was already there?

**Comment by esoubiran-aneo** [comment]
I  tried to fix the more issue more I don't understand every part of the code and links between files;

File: armonik/admin-0-9-gui.tf [hunk 9]
```diff
@@ -0,0 +1,153 @@
+# Control plane deployment
+resource "kubernetes_deployment" "admin_0_9_gui" {
+  count = var.admin_gui != null ? 1 : 0
```
Comment (esoubiran-aneo) [inline armonik/admin-0-9-gui.tf:3]: Do I need to update this?

Comment (lemaitre-aneo) [reply armonik/admin-0-9-gui.tf:3]: Yes, it should be 
```suggestion
  count = var.admin_0_9_gui != null ? 1 : 0
```

**Review by lemaitre-aneo — APPROVED** [review]

**Pull request merged**